examples/bri-3/src/bri/zeroKnowledgeProof/agents/ccsmAnchorHash.agent.ts

```diff
@@ -0,0 +1,48 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { INVALID_ANCHOR_HASH_INPUT } from '../api/err.messages';
+import { v4 as uuidv4 } from 'uuid';
+import { CCSMAnchorHash } from '../models/ccsmAnchorHash';
+
+@Injectable()
+export class CCSMAnchorHashAgent {
+  public throwErrorIfCCSMAnchorHashInputInvalid(
+    inputForProofVerification: string,
+  ): void {
+    if (!inputForProofVerification) {
+      throw new BadRequestException(INVALID_ANCHOR_HASH_INPUT);
+    }
+  }
+
+  public createNewCCSMAnchorHash(
+    ownerId: string,
+    document: string,
+  ): CCSMAnchorHash {
+    const hash = this.convertDocumentToHash(document);
+
+    return new CCSMAnchorHash(uuidv4(), ownerId, hash);
+  }
+
+  public verifyCCSMAnchorHash(
+    CCSMAnchorHash: string,
+    publicInputForProofVerification: string,
+  ): boolean {
+    if (CCSMAnchorHash === publicInputForProofVerification) {
```

Comment:
this if is redundant you can just go with `return CCSMAnchorHash === publicInputForProofVerification`